.jenkins/common.groovy

```diff
@@ -23,170 +23,102 @@
+            make -j\$((`nproc`<16 ? `nproc` : 16))
 
             popd
             """
 
-    try
-    {
-        platform.runCommand(this, command)
-    }
-    catch(e)
-    {
-        try
-        {
-            junit "${project.paths.project_build_prefix}/python_unit_tests.xml"
-        }
-        catch(ee)
-        {}
-
-        throw e
-    }
-
-    junit "${project.paths.project_build_prefix}/python_unit_tests.xml"
+    platform.runCommand(this, command)
 }
 
-def publishResults(project, boolean skipHostTest=false)
-{
-    try
-    {
-        archiveArtifacts "${project.paths.project_build_prefix}/timing*.csv"
-    }
-    finally
-    {
-        try
-        {
-            if (!skipHostTest) junit "${project.paths.project_build_prefix}/build/host_test_output.xml"
-        }
-        finally
-        {
-            junit "${project.paths.project_build_prefix}/python_tests.xml"
-        }
-    }
-}
 
-def runTestCommand (platform, project, jobName, test_marks, boolean skipHostTest=false)
+def runTestCommand(platform, project, jobName, testMark, boolean runHostTest=true, boolean runUnitTest=true)
 {
-    def test_dir =  "Tensile/Tests"
-
     String compiler = '/opt/rocm/bin/amdclang++'
-    String pythonVersion = 'py3'
-    String markSkipHostTest = skipHostTest ? "#" : ""
-    String markSkipExtendedTest = !test_marks.contains("extended") ? "\"--gtest_filter=-*Extended*:*Ocl*\"" : "\"--gtest_filter=-*Ocl*\""
+    String markSkipExtendedTest = !testMark.contains("extended") ? "\"--gtest_filter=-*Extended*:*Ocl*\"" : "\"--gtest_filter=-*Ocl*\""
 
     def command = """#!/usr/bin/env bash
-            set -x
+            check_err() {
+              local ERR=\$?
+              if [ \$ERR -ne 0 ]; then
+                exit \$ERR
+              fi
+            }
 
+            set -x
             hostname
-
-            export PATH=/opt/rocm/bin:\$PATH
+            date
             cd ${project.paths.project_build_prefix}
 
-            gpuArch=`/opt/rocm/bin/rocm_agent_enumerator  | tail -n 1`
-
-            ${markSkipHostTest}pushd build
-            ${markSkipHostTest}./TensileTests ${markSkipExtendedTest} --gtest_output=xml:host_test_output.xml --gtest_color=yes
-            ${markSkipHostTest}HOST_ERR=\$?
-            ${markSkipHostTest}popd
-
-            #### temporary fix to remedy incorrect home directory
             export HOME=/home/jenkins
-            ####
-            tox --version
+            export PATH=/opt/rocm/bin:\${PATH}
             export TENSILE_COMPILER=${compiler}
-            tox -v --workdir /tmp/.tensile-tox -e ${pythonVersion} -- ${test_dir} -m "${test_marks}" --timing-file=\$(pwd)/timing-\$gpuArch.csv
-            PY_ERR=\$?
-            date
+            export GPU_ARCH=`/opt/rocm/bin/rocm_agent_enumerator  | tail -n 1`
+            export TIMING_FILE=`pwd`/timing-\${GPU_ARCH}.csv
+
+            tox --version
+            tox run -e ci -- -m ${testMark} --timing-file=\${TIMING_FILE} --numprocesses=4
```

Comment:
Some of the process/thread code got re-arranged. Looks like it's all the same settings, but would be a good idea to double check the output on the CI run for this PR. If threading is too high, we get test failures due to the RAM on CI nodes being shared across multiple jobs. If the threading is too low, we sometimes hit the timeout limit (particularly for extended tests). Would be worth double checking that the test run-time is similar to previous CI jobs.

Reply:
I noticed this behaviour, originally, I set it to **auto** (which sets defaults to nproc) thinking that it would speed up our CI jobs but ran into the issues you're describing. I've updated the base numprocesses to 8 in *tox.ini*, so this will be used when running locally with `tox run -e ci ...`, but override that to 4 in CI, which it what it was set to before these changes---seems to be the sweet spot.

Reply:
I've noted some non-determinism with the setup I was using so I defaulted everything back to 4 processes (how it was before) and the tests now appear to be reliably passing.